### DISCUSSION TITLE
How to get result of regression?

### DISCUSSION THREAD

**Discussion by fundef1** [observablehq/framework#1786] [Q&A] [answered]
Hi, 
how would I get the result of a linearRegressionY plot in framework?
e.g. a function(ts) return val , or the params of the line drawn... is this exposed somehow?
ideally I can get this in the title (like with bin)

quick mockup: 
```
---
title: Random Regression
---

# test regression

~~~js --reminder: no semicolon or framework won't render
Plot.plot({
    x: {type: "utc"},
    title: "how to get result of regression here?",
    marks: [
        Plot.linearRegressionY(
            await sql`SELECT generate_series AS ts, trunc(random()*10)::INT AS val FROM generate_series(DATE '2022-01-01', DATE '2024-01-01', INTERVAL '1' MONTH);`, 
            { x: "ts", y: "val", title: "or maybe here?" })
    ]
})
~~~

thanks!
```

**Comment by Fil** [comment]
Plot is not replacing a statistics package—this mark only does linear regression for the sake of visualization, and does not expose its coefficients. (And in fact, it computes its parameters in pixel space, so it would not be very useful to expose them anyway. Only the coefficient of determination R² would be what you expect, if the scales are linear.)

We recommend using specialized packages when you want to compute statistics. There are plenty of them, check out for example https://simple-statistics.github.io/ and https://github.com/nshiab/simple-data-analysis

I can share examples of how to import them in Observable Framework.

**Comment by fundef1** [comment, marked as answer]
alright, very clear. thanks.
But in that case I might as  well pull the regression from DuckDB (where the data lives anyway) instead of pulling in another lib in javascript. (although it did get fidgety dealing with the timestamps)
```
~~~js 
await sql`CREATE OR REPLACE TABLE byidx AS SELECT generate_series AS idx, trunc(random()*10)::INT AS val FROM generate_series(1, 25, 1)`;
const byidx = await sql `FROM byidx`;
const [{slope,intercept}] = await sql`SELECT regr_slope(val,idx) AS slope, regr_intercept(val,idx) as intercept FROM byidx`;
~~~ 
~~~js 
    Plot.plot({
    title: `val = idx * ${slope.toFixed(2)} + ${intercept.toFixed(2)}`,
    marks: [
        Plot.lineY(byidx, {stroke: "red", x: "idx", y: (d) => d.idx*slope+intercept-1}), //-1 not to overplot
        Plot.linearRegressionY(byidx, { x: "idx", y: "val" })
    ]
})
~~~
```

thanks!
